presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpiller.java

```diff
@@ -99,7 +107,8 @@ private void checkNoSpillInProgress()
 
     private void writePages(Iterator<Page> pageIterator)
     {
-        try (SliceOutput output = new OutputStreamSliceOutput(new FileOutputStream(targetFileName.toFile(), true))) {
+        try (SliceOutput output = new OutputStreamSliceOutput(new FileOutputStream(targetFileName.toFile(), true), BUFFER_SIZE)) {
+            memoryContext.setBytes(BUFFER_SIZE);
```

Comment:
In this method, you definitely should not... there are tons of methods in Guava we call that allocate 4k buffers and toss them.